app/src/main/java/org/thoughtcrime/securesms/database/RecipientSettingsDatabase.kt

```diff
@@ -143,6 +145,88 @@ class RecipientSettingsDatabase @Inject constructor(
         }
     }
 
+    /**
+     * This method returns all profile pic url and key.
+     * This will be used to identify which avatars are still being used  to exclude
+     * them from the cleanup.
+     */
+    fun getAllReferencedAvatarFiles(): Set<RemoteFile.Encrypted> {
+        val recipientAvatars = LinkedHashSet<RemoteFile.Encrypted>()
+        readableDatabase.rawQuery(
+            """
+        SELECT $COL_PROFILE_PIC_URL, $COL_PROFILE_PIC_KEY
+        FROM $TABLE_NAME
+        WHERE $COL_PROFILE_PIC_URL IS NOT NULL AND $COL_PROFILE_PIC_URL != ''
+          AND $COL_PROFILE_PIC_KEY IS NOT NULL AND $COL_PROFILE_PIC_KEY != ''
+        """.trimIndent(),
+            null
+        ).use { cursor ->
+            while (cursor.moveToNext()) {
+                val url = cursor.getString(0)
+                val keyB64 = cursor.getString(1)
+                runCatching {
+                    val keyBytes = Base64.decode(keyB64)
+                    recipientAvatars += RemoteFile.Encrypted(url = url, key = Bytes(keyBytes))
+                }.onFailure {
+                    // ignore bad rows
+                }
+            }
+        }
+        return recipientAvatars
+    }
+
+    fun getAllRecipientAddresses(): Set<Address> {
+        return readableDatabase.rawQuery(
+            "SELECT $COL_ADDRESS FROM $TABLE_NAME", emptyArray()
```

Comment:
I think you don't need the `emptyArray()`
```suggestion
            "SELECT $COL_ADDRESS FROM $TABLE_NAME"
```